### ISSUE TITLE
Setup random seed for multi_gpu_model

### ISSUE THREAD

**Issue by shu-hai** [keras-team/keras#9782]
Hi,
I used the following code to set up random seeds, but "multi_gpu_model" gives different results every time. So, I was wondering how to set up random seeds to assure reproducible work? I use Tensorflow as the backend.

```
np.random.seed(0)
from tensorflow import set_random_seed
set_random_seed(0)
import random
random.seed(0)
os.environ['PYTHONHASHSEED'] = '0'
```

**Comment by TristanJM** [comment]
Hey,
I was getting a similar problem.

Following the docs (https://keras.io/utils/#multi_gpu_model) fixed the problem for me:

Make sure you create and compile the base template model on your CPU rather than a GPU. I'm guessing it then uses the same seeded randoms to initiate the model weights rather than loading unpredictably on a GPU.

It will now reproduce the multi_gpu results provided I did not change the number of GPUs it runs on.

```python
with tf.device('/cpu:0'):
    model = # etc etc
    model.compile(loss='mse' #, ...)

parallel_model = multi_gpu_model(model, gpus = 4)
parallel_model.compile(loss='mse' #, ...)

parallel_model.fit()

model.save()
```
